head.go

```diff
@@ -954,7 +954,7 @@ func (h *Head) gc() {
 	h.postings.Delete(deleted)
 
 	// Rebuild symbols and label value indices from what is left in the postings terms.
-	symbols := make(map[string]struct{})
+	symbols := make(map[string]struct{}, len(h.values)*2)
```

Comment:
Why not use h.symbols? This is also only called every 2 hours, so won't make much of a difference.

Reply:
Thanks @brian-brazil 

Usually, the lengths of new `symbols` and `h.symbols` are similar.
So might be worthwhile to allocate a map of size `len(h.symbols)`.

But as you said,  this is only called every 2 hours (or delete request from admin API), won't make much of a difference.

( If this change is not needed, I will close this pr. )